File: DataStructures/Arrays/Arrays-DS.scala
object Solution {

    def main(args: Array[String]) {
        val sc = new java.util.Scanner (System.in);
        var n = sc.nextInt();
        var arr = new Array[Int](n);
        for(arr_i <- 0 to n-1) {
           arr(arr_i) = sc.nextInt();
        }

        val list = arr.toList

        def reverse(ls: List[Int]): String = ls match {
          case Nil      => ""
          case x :: Nil => x.toString()
          case x :: xs  => reverse(xs) + " " + x
        }

        println(reverse(list))
    }
}

File: DataStructures/Arrays/Arrays_DS.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){
  int n;
  cin >> n;
  vector<int> arr(n);
  for(int i = 0; i < n; ++i){
    cin >> arr[i];
  }

  for(int i = n; i > 0; --i){
    cout << arr[i-1] << " ";
  }

  return 0;
}
